java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1393,8 +1406,21 @@ private void handleAppStreamsChange(ServerAction serverAction, JsonElement jsonR
         if (server.isEmpty()) {
             return;
         }
-        var currentlyEnabled = Json.GSON.fromJson(jsonResult, AppStreamsChangeSlsResult.class).getCurrentlyEnabled();
 
+        if (ActionFactory.STATUS_FAILED.equals(serverAction.getStatus())) {
+            // Filter out the subsequent errors to find the root cause
+            var originalErrorMsg = jsonEventToStateApplyResults(jsonResult)
+                    .map(SaltUtils::getOriginalStateApplyError)
+                        .orElseThrow(() -> new RuntimeException("Failed to parse the state.apply error result"))
```

Comment:
This one is actually impossible to reach in normal circumstances, as long as the result belongs to a `state.apply` call, which always is in our case, but `findFirst` in the stream API returns an Optional. I added the `RuntimeException` here because if it happens something is weird and we should look into it.